Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -363,20 +363,18 @@ PhysicalParticleContainer::AddPlasmaFromFile(ParticleReal q_tot,
             ParticleReal const z = ptr_z.get()[i]*position_unit_z+z_shift;
 #   ifndef WARPX_DIM_3D
             ParticleReal const y = 0.0_prt;
+            ParticleReal const uy = 0.0_prt;
 #   else
             ParticleReal const y = ptr_y.get()[i]*position_unit_y;
+            ParticleReal const uy = ptr_uy.get()[i]*momentum_unit_y/PhysConst::m_e;
```

Comment:
You now run into the risk that `uy` is not defined in a 2D input file and `ptr_uy` is a `nullptr`. I would just suggest to keep the logic that is below with
```C++
                ParticleReal uy = 0.0_prt;
                if (ps["momentum"].contains("y")) {
                    uy = ptr_uy.get()[i]*momentum_unit_y/PhysConst::m_e;
                }
```

and don't mix position and momentum component loading here.

Reply:
I think you read the `#ifdef` wrong. Those 2 lines are read only when we run in 3D

Reply:
In that case we assume that openPMD will crash the run if any component is missing (we say which ones are required in the Docs) and then we get the error message that the component is missing.

Reply:
In 2D we set `y` and `uy` to 0.